frontend/integration-tests/tests/source-to-image.scenario.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import { browser, $, element, by, ExpectedConditions as until } from 'protractor';
+import * as _ from 'lodash';
+
+import { appHost, checkLogs, checkErrors, testName } from '../protractor.conf';
+import * as crudView from '../views/crud.view';
+import * as sourceToImageView from '../views/source-to-image.view';
+
+describe('Source-to-Image', () => {
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  describe('Image stream page', () => {
+    it('has Create Application button for builder image', async() => {
+      await sourceToImageView.visitOpenShiftImageStream('nodejs');
+      expect(sourceToImageView.createApplicationButton.isPresent()).toBe(true);
+    });
+
+    it('does not have Create Application button for non-builder image', async() => {
+      await sourceToImageView.visitOpenShiftImageStream('jenkins');
+      expect(sourceToImageView.createApplicationButton.isPresent()).toBe(false);
+    });
+  });
+
+  describe('Node.js app', () => {
+    const appName = 'test-nodejs';
+    const appLabel = `app=${appName}`;
+    const resources = {
+      'buildconfigs': {kind: 'BuildConfig'},
+      'deploymentconfigs': {kind: 'DeploymentConfig'},
+      'imagestreams': {kind: 'ImageStream'},
+      'routes': {kind: 'Route'},
+      'services': {kind: 'Service'},
+    };
+
+    const visitResource = (resource: string) => {
```

Comment:
Should be an `async` function that returns a promise.

Reply:
I don't think `browser.get()` returns a promise

Reply:
It does ([source](https://github.com/angular/protractor/blob/master/lib/browser.ts#L859)) and we [usually write `await browser.get()`](https://github.com/openshift/console/blob/master/frontend/integration-tests/tests/crud.scenario.ts#L71).

Reply:
OK thanks, updated